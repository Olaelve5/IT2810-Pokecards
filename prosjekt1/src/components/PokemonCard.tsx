import classes from '../styles/PokemonCard.module.css';
import { useDataContext } from '../contexts/DataContext';
import FavouriteButton from './buttons/FavouriteButton';

const Pokemon = () => {
  const { activePokemon } = useDataContext();

  if (!activePokemon) {
    return <div className={classes.container}>No Pokemon selected</div>;
  }

  return (
    <div className={classes.container}>
      <div className={classes.name}>{activePokemon.name}</div>
      {activePokemon.images.large && <img src={activePokemon.images.large} alt={activePokemon.name} className={classes.image}/>}
      <div className={classes.favoriteIdContainer}>
        <div className={classes.id}>#{activePokemon.number}</div>
        <FavouriteButton pokemonId={''} />
      </div>
    </div>
  );
};

export default Pokemon;
