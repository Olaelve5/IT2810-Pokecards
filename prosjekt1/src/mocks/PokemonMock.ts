const pokemon1 = {
  number: 1,
  name: 'Beedrill',
  images: {
    small: 'https://example.com/small.png',
    large: 'https://example.com/large.png',
  },
};

const pokemon2 = {
  number: 2,
  name: 'Butterfree',
  images: {
    small: 'https://example.com/small.png',
    large: 'https://example.com/large.png',
  },
};

const pokemon3 = {
  number: 3,
  name: 'Pikachu',
  images: {
    small: 'https://example.com/small.png',
    large: 'https://example.com/large.png',
  },
};

export const tablePokemonsMock = [pokemon1, pokemon2, pokemon3];
