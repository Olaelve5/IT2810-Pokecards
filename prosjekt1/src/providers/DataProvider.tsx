import { useEffect, useState } from "react";
import { DataContext } from "../contexts/DataContext";
import { PokemonType } from "../types/Pokemon";

const DataProvider = ({ children }: { children: React.ReactNode }) => {
    const [activePokemon, setActivePokemon] = useState<PokemonType | null>(null);
    const [tablePokemons, setTablePokemons] = useState<PokemonType[]>([]);
    const [showFavorites, setShowFavorites] = useState<boolean>(false);

    useEffect(() => {
        if (!activePokemon && tablePokemons.length > 0) {
            setActivePokemon(tablePokemons[0]);
        }
    }, [activePokemon, tablePokemons]);

    return (
        <DataContext.Provider value={{ 
            activePokemon, setActivePokemon, 
            tablePokemons, setTablePokemons,
            showFavorites, setShowFavorites
            }}>
            {children}
        </DataContext.Provider>
    );
};

export default DataProvider;

