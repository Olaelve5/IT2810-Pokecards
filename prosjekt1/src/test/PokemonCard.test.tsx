import { describe, it, vi, expect } from 'vitest';
import { render, screen } from '@testing-library/react';
import PokemonCard from '../components/PokemonCard';
import DataProvider from '../providers/DataProvider';
import { tablePokemonsMock } from '../mocks/PokemonMock';

// Test if the PokemonCard component renders the active pokemon
describe('PokemonCard', () => {
  beforeEach(() => {
    vi.mock('../contexts/DataContext', async () => {
      const actual = await import('../contexts/DataContext');
      return {
        ...actual,
        useDataContext: () => ({
          activePokemon: tablePokemonsMock[0],
          tablePokemons: tablePokemonsMock,
        }),
      };
    });
  });

  afterEach(() => {
    vi.restoreAllMocks();
  });

  it('renders pokemon', async () => {
    const pokemonCard = render(
      <DataProvider>
        <PokemonCard />
      </DataProvider>,
    );

    expect(pokemonCard).toMatchSnapshot();

    // Test if mockPokemon1 is rendered correctly
    expect(screen.getByText(tablePokemonsMock[0].name)).toBeInTheDocument();

    // Test if the image of mockPokemon1 is rendered correctly
    const image = screen.getByAltText(`${tablePokemonsMock[0].name} image`);
    expect(image).toBeInTheDocument();
    expect(image).toHaveAttribute('src', tablePokemonsMock[0].images.large);
  });
});
