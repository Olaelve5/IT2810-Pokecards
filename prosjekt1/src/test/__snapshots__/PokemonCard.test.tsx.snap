// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PokemonCard > renders pokemon 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="_container_836431"
      >
        <div
          class="_name_836431"
          data-testid="displayName"
        >
          Beedrill
        </div>
        <img
          alt="Beedrill image"
          class="_image_836431"
          src="https://example.com/large.png"
        />
        <div
          class="_favoriteIdContainer_836431"
        >
          <div
            class="_id_836431"
          >
            #
            1
          </div>
          <button
            class="_favouriteButton_e808fb"
          >
            <svg
              class="tabler-icon tabler-icon-star "
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 17.75l-6.172 3.245l1.179 -6.873l-5 -4.867l6.9 -1l3.086 -6.253l3.086 6.253l6.9 1l-5 4.867l1.179 6.873z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="_container_836431"
    >
      <div
        class="_name_836431"
        data-testid="displayName"
      >
        Beedrill
      </div>
      <img
        alt="Beedrill image"
        class="_image_836431"
        src="https://example.com/large.png"
      />
      <div
        class="_favoriteIdContainer_836431"
      >
        <div
          class="_id_836431"
        >
          #
          1
        </div>
        <button
          class="_favouriteButton_e808fb"
        >
          <svg
            class="tabler-icon tabler-icon-star "
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 17.75l-6.172 3.245l1.179 -6.873l-5 -4.867l6.9 -1l3.086 -6.253l3.086 6.253l6.9 1l-5 4.867l1.179 6.873z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
